File: trunk/trunk/PHPAspect/Runtime/Reflection/JoinPoint.php
<?php

require_once 'Runtime/Reflection/InvalidJoinPointException.php';

interface JoinPoint extends Reflector{

	const CONSTRUCTOR_CALL = 'constructor-call';

	const METHOD_CALL = 'method-call';

	const METHOD_EXECUTION = 'method-execution';

	const FIELD_GET = 'field-get';

	const FIELD_SET = 'field-set';

	const EXCEPTION_HANDLER = 'exception-handler';

	public function __toString();

	public function toString();

	public function toLongString();

	public function toShortString();

	public function getSignature();

	public function getThis();

	public function getSource();

	public function getTarget();

	public function getFileName();

	public function getLine();

	public function getKind();
}
?>

File: trunk/org.phpaspect.weaver/src/org/phpaspect/weaver/JoinPoint.java
package org.phpaspect.weaver;

import org.eclipse.php.internal.core.ast.nodes.ASTNode;

public interface JoinPoint {

	public enum Kind { CONSTRUCTOR_CALL, CONSTRUCTOR_EXECUTION, EXCEPTION_HANDLER,
                       FIELD_GET, FIELD_SET, METHOD_CALL, STATIC_METHOD_CALL, METHOD_EXECUTION };

    public Kind getKind();
 
    public SourceLocation getSourceLocation();

    public ASTNode getNode();

    public String toLongString();

    public String toString();

    public String toShortString();
}
